settings.gradle

```diff
@@ -38,9 +38,31 @@ rootProject.children.each { subproject ->
 
 
 buildCache {
+
+    def getFile = { dir, filename ->
+        File file = new File("$dir$File.separator$filename")
+        file?.exists() ? file : null
+    }
+
+    def getLocalProperties = { dir ->
+        def file = getFile(dir, "local.properties")
+        if (!file) {
+            return null
+        }
+
+        Properties properties = new Properties()
+        properties.load(file.newInputStream())
+        return properties
+    }
+
     local {
+        def properties = getLocalProperties(rootDir)
+        if (properties != null) {
+            enabled = "true" == properties.getProperty("local.build.cache", "true")
+        } else {
+            enabled = true
```

Comment:
As per top-level comment, this makes the build cache **enabled by default**. I think we're better making it disabled by default, but no strong opinions really